### DISCUSSION TITLE
Wezterm keybinding & terminal keybinding: Conditional switch?

### DISCUSSION THREAD

**Discussion by chrisgrieser** [wezterm/wezterm#4101] [Q&A] [answered]
As far as I can tell, WezTerm comes first in the stack, meaning that when a certain keybinding is mapped in the shell (e.g. via `bindkey` in zsh) and mapped in my WezTerm config, the WezTerm keybinding is prioritized.

Now, is it possible to conditionally switch between them? In particular, I want to bind `PageUp` and `PageDown` to scroll results in `fzf`, but if `fzf` is not running, scroll the WezTerm window. Is something like that somehow possible?

**Comment by aj3423** [comment, marked as answer]
This is how I conditionally bind `PageUp` to scroll up in wezterm and pass the event to `nvim` if it's running in foreground:
```lua
		{
			key = 'PageUp',
			action = wezterm.action_callback(function(win, pane)
				if pane:is_alt_screen_active() then
					win:perform_action(wezterm.action.SendKey{ key = 'PageUp' }, pane)
				else
					win:perform_action(wezterm.action.ScrollByPage(-1), pane)
				end
			end)
		},

```

**Reply by chrisgrieser** [reply]
thank you, that one does it!

Has the caveat of requiring `--height=100%` for `fzf`, since with a lower height, `fzf` apparently does not create such an alternate screen. But better than nothing

**Reply by aj3423** [reply]
Maybe try comparing the foreground process name:
```lua
action = wezterm.action_callback(function(win, pane)
  local name = basename(pane:get_foreground_process_name())
  if name == "fzf" then
    ...   
  else
    ...
  end
end)
```

**Reply by chrisgrieser** [reply]
That one I actually already tried. The issue here is that it works when you simply call `fzf` in the terminal, but when you call `fzf` in a function of yours with subshells, then it still returns something like `/bin/zsh`.
